modules/tinymce/src/core/test/ts/browser/keyboard/PageUpDownKeysTest.ts

```diff
@@ -0,0 +1,88 @@
+import { Keys } from '@ephox/agar';
+import { context, describe, it } from '@ephox/bedrock-client';
+import { TinyAssertions, TinyContentActions, TinyHooks, TinySelections } from '@ephox/mcagar';
+import { PlatformDetection } from '@ephox/sand';
+
+import Editor from 'tinymce/core/api/Editor';
+import Theme from 'tinymce/themes/silver/Theme';
+
+describe('browser.tinymce.core.keyboard.PageUpDownKeyTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    add_unload_trigger: false,
+    base_url: '/project/tinymce/js/tinymce',
+    indent: false
+  }, [ Theme ], true);
+  const platform = PlatformDetection.detect();
+
+  context('Page Up', () => {
+    it('TINY-4612: caret should be placed before the inline element if it is at the start of it', () => {
+      const editor = hook.editor();
+      editor.setContent('<p>text<a href="google.com">link</a></p>');
+      TinySelections.setCursor(editor, [ 0, 1, 0 ], 0);
+      TinyContentActions.keyup(editor, Keys.pageUp());
+
+      if (platform.browser.isIE()) {
+        TinyAssertions.assertCursor(editor, [ 0, 0 ], 4);
+      } else {
+        TinyAssertions.assertCursor(editor, [ 0 ], 1);
+      }
+    });
+
+    it('TINY-4612: caret should be placed before the inline element if it is at the end of it', () => {
```

Comment:
This text and also the behaviour by the looks isn't correct, as the link is in the middle of the content. So in this case, pageup shouldn't be doing anything and should be falling back to the browser behaviour (ie should be before `text`).

Reply:
Just leaving another comment to make sure it hasn't been lost. This doesn't appear to have been addressed yet, as the tests still have the wrong assertions. Remember that Page Up/Down should go to the start/end of the line in the viewport not to the start of the anchor, when the anchor is in the middle of a line.

Also worth noting, that @JamesToohey found out yesterday that Page Up/Down on Mac appears to scroll the viewport instead of change the caret position/selection.

Reply:
The current behavior for PageUp/PageDown is to move the caret outside the inline element if it is at the start/end of it without overwriting default behavior, not move the caret at the start/end of the line (that it's a browser behavior that only happens on Windows os)

Reply:
And that's what I'm saying is incorrect behaviour. It should **never** move it outside the anchor unless the anchor is at the start/end of the line. Also, the behaviour I mentioned happens on every OS except macOS in our testing, so saying it only applies to Windows would be incorrect.